### DISCUSSION TITLE
[@types/qs] wrong ParsedQs type

### DISCUSSION THREAD

**Discussion by vovochka-dev** [DefinitelyTyped/DefinitelyTyped#71254] [Issues with a @types package]
Error in this code:
```
import qs from 'qs';

const queryString = 'user[0]=Alice&user[1][name]=Bob';
const parseResult = qs.parse(queryString); // { user: [ 'Alice', { name: 'Bob' } ] }

// error Type 'string' is not assignable to type 'ParsedQs'
const testType: qs.ParsedQs = { user: ['Alice', { name: 'Bob' }] };
```

Problem resolves if replace this:
```
interface ParsedQs {
    [key: string]: undefined | string | string[] | ParsedQs | ParsedQs[];
}
```
to this:
```
interface ParsedQs {
   [key: string]: undefined | string | ParsedQs | (string | ParsedQs)[];
}
```

[a comment by an automated account is not shown]

**Comment by ljharb** [comment]
`string[] | ParsedQs[]` means, an array of all strings, or an array of all ParsedQs.

`(string | ParsedQs)[]` means, an array that has all strings, or all ParsedQs, or a mixture of both.

I think this is a correct change. Want to make a PR?

**Comment by duncanbeevers** [comment]
Opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71581 to address this issue.

**Comment by duncanbeevers** [comment]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71581 has landed and addresses this issue.

I think this discussion can be closed.
